pandas/core/window.py

```diff
@@ -246,8 +246,11 @@ def _prep_values(self, values: Optional[np.ndarray] = None) -> np.ndarray:
             except (ValueError, TypeError):
                 raise TypeError("cannot handle this type -> {0}".format(values.dtype))
 
-        # Always convert inf to nan
-        values[np.isinf(values)] = np.NaN
+        # Convert inf to nan for C funcs
+        inf = np.isinf(values)
+        if inf.any():
+            values = values.copy()  # GH-27766, Don't write into user's data
```

Comment:
comment on previous line.

is the issue that we dont want to overwrite user's data, or that the user might pass read-only array?

Reply:
Mmm, actually yeah, it's the first.